src/derive/curve.rs

```diff
@@ -132,58 +192,41 @@ macro_rules! new_curve_impl {
                     }
 
                     fn from_bytes_unchecked(bytes: &Self::Repr) -> CtOption<Self> {
+                        // In compressed form we cannot skip the curve check.
                         Self::from_bytes(bytes)
                     }
 
                     fn to_bytes(&self) -> Self::Repr {
                         if bool::from(self.is_identity()) {
```

Comment:
The `from_bytes` is constant time, but here we branch based on `is_identity` which makes the `to_bytes` not constant time.
This seems inconsistent.